src/App.tsx

```diff
@@ -16,36 +15,29 @@ export const App = () => (
       data-cy="Nav"
     >
       <div className="container">
-        <div className="navbar-brand">
-          <a href="/" className="navbar-item is-active">Home</a>
-          <a href="/tabs" className="navbar-item">Tabs</a>
-        </div>
+        <Navigation />
       </div>
     </nav>
 
     <div className="section">
       <div className="container">
-        <h1 className="title">Home page</h1>
-        <h1 className="title">Tabs page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="tabs is-boxed">
-          <ul>
-            <li data-cy="Tab" className="is-active">
-              <a href="#/">Tab 1</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 2</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 3</a>
-            </li>
-          </ul>
-        </div>
-
-        <div className="block" data-cy="TabContent">
-          Please select a tab
-        </div>
+        <Routes>
+          <Route path="/" element={<h1 className="title">Home page</h1>} />
+          <Route path="home" element={<Navigate to="/" replace />} />
+          <Route
+            path="tabs"
+            element={(
+              <>
+                <h1 className="title">Tabs page</h1>
+                <Outlet />
+              </>
+            )}
+          >
+            <Route index element={<TabPage />} />
+            <Route path=":tabId" element={<TabPage />} />
+          </Route>
+          <Route path="*" element={<h1 className="title">Page not found</h1>} />
+        </Routes>
```

Comment:
use separeate component for Page not found is better practice. Especially if you will need add somethink on this page (picture, for example).